boards/arm/nucleo_l4r5zi/doc/nucleol4r5zi.rst

```diff
@@ -0,0 +1,252 @@
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c                                 |
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi                                 |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/nucleo_l4r5zi/nucleo_l4r5zi_defconfig``
+
+
+Connections and IOs
+===================
+
+Nucleo L4R5ZI Board has 8 GPIO controllers. These controllers are
+responsible for pin muxing, input/output, pull-up, etc.
+
+Available pins:
+---------------
+.. image:: img/nucleo144_layout.jpg
+     :width: 720px
+     :align: center
+     :height: 540px
+     :alt: Nucleo L4R5ZI Arduino connectors
+
+For mode details please refer to `STM32 Nucleo-144 board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_1_TX : PA9
+- UART_1_RX : PA10
+- UART_2_TX : PA2
+- UART_2_RX : PA3
+- UART_3_TX : PB10
+- UART_3_RX : PB11
+- I2C_1_SCL : PB6
+- I2C_1_SDA : PB7
+- SPI_1_NSS : PA4
+- SPI_1_SCK : PB3
+- SPI_1_MISO : PA6
+- SPI_1_MOSI : PA7
+- SPI_2_NSS : PB12
+- SPI_2_SCK : PB13
+- SPI_2_MISO : PB14
+- SPI_2_MOSI : PB15
+- SPI_3_NSS : PB12
+- SPI_3_SCK : PC10
+- SPI_3_MISO : PC11
+- SPI_3_MOSI : PC12
+- PWM_2_CH1 : PA0
+- USER_PB : PC13
+- LD1 : PC7
+- LD2 : PB7
+- LD3 : PB14
+
+System Clock
+------------
+
+Nucleo L4R5ZI System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default System clock is driven by PLL clock at 120MHz,
+driven by 16MHz high speed internal oscillator.
+
+Serial Port
+-----------
+
+Nucleo L4R5ZI board has 5 U(S)ARTs. The Zephyr console output is assigned to UART2.
+Default settings are 115200 8N1.
+
+
+Programming and Debugging
+*************************
+
+Applications for the ``nucleo_l4r5zi`` board configuration can be built and
+flashed in the usual way (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Flashing
+========
+
+Nucleo L4R5ZI board includes an ST-LINK/V2-1 embedded debug tool interface.
+
+You currently need bleeding edge OpenOCD plus a few patches (build your own OpenOCD):
```

Comment:
bleeding edge?  better would be to say which version (or greater) is needed.  Are the two bullet points below sufficient instructions to build a working OpenOCD?  If not, please provide instructions.